packages/grafana-prometheus/src/query_hints.ts

```diff
@@ -230,6 +210,126 @@ export function getInitHints(datasource: PrometheusDatasource): QueryHint[] {
   return hints;
 }
 
+export function getExpandRulesHints(query: string, mapping: RuleQueryMapping): QueryHint[] {
+  const hints: QueryHint[] = [];
+  const mappingForQuery = Object.keys(mapping).reduce((acc, ruleName) => {
+    if (query.search(ruleName) === -1) {
+      return acc;
+    }
+
+    if (mapping[ruleName].length > 1) {
+      const { idx, expandedQuery, identifier, identifierValue } = getRecordingRuleIdentifierIdx(
+        query,
+        ruleName,
+        mapping[ruleName]
+      );
+
+      // No identifier detected add warning
+      if (idx === -1) {
+        hints.push({
+          type: 'EXPAND_RULES_WARNING',
+          label:
+            'We found multiple recording rules that match in this query. To expand the recording rule, add an identifier label/value.',
```

Comment:
When a hint does not have a fix or action it won't show up in the query builder. This warning is only shown in the code editor.

https://github.com/grafana/grafana/blob/1404249ff151839bb91bc2266e058fd8ffee9e97/packages/grafana-prometheus/src/querybuilder/shared/QueryBuilderHints.tsx#L36

Reply:
In the future we will have an action for that warning. But for now, that's how it is. And there is a line above the one you linked, it says `For now show only actionable hints` so I think this is fine for now.